auto_refresh_token_provider.go

```diff
@@ -0,0 +1,25 @@
+package incognia
+
+type AutoRefreshTokenProvider struct {
+	manualRefreshTokenProvider *ManualRefreshTokenProvider
+}
+
+func NewAutoRefreshTokenProvider(tokenClient *TokenClient) *AutoRefreshTokenProvider {
+	return &AutoRefreshTokenProvider{manualRefreshTokenProvider: NewManualRefreshTokenProvider(tokenClient)}
+}
+
+func (t *AutoRefreshTokenProvider) GetToken() (*AccessToken, error) {
+	accessToken, err := t.manualRefreshTokenProvider.GetToken()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
When running for the first time, the token will initially be nil. This will return an error in `ManualRefreshTokenProvider.GetToken`, which will be caught here and the code will never proceed to refresh it. Maybe the `GetToken` method from `ManualRefreshTokenProvider` should not return an error, but only an `*AccessToken`.

Reply:
+1

Reply:
This will break the interface :thinking:

Reply:
Instead of doing that, I just separated the tokens providers completely, so we can keep the interface between them.